### PR TITLE
Fix: Use popper instead of portal for meatball menu

### LINKED ISSUE #3579: QA Dashboard: Mobile menu doesn't sticky on scroll
## Steps to reproduce

1. On mobile, expand a row within with the dashboard's activity feed.
2. Scroll down and watch the dropdown menu move with the scroll and cover the UI.
3. Close the menu.

## Expected behaviour

* The meatball menu on mobile triggered from a row in the activity feed should 'stick' to the meatball menu on scroll. 

1. Scroll up to the 'Team rep' widget.
2. Select the kebab icon.
3. Scroll up and down. The menu will not move on scroll and 'stick' to the correct position.

## Actual behaviour

The meatball menu moves on scroll covering up the UI and activity feed information.

* [Example: A `WALLET_CREATE_ERROR` action is dispatched and the user cannot log in](https://github.com/user-attachments/assets/276efacc-0958-4540-bd79-ac81a2e84b8b)

### PR DESCRIPTION
**Pull request by mmioana**
## Description

- This PR ensures the meatball menu remains sticky to its parent upon scrolling.

## Testing

TODO: Check the changes for the meatball menu component trigger the expected behaviour and doesn't break between viewport resizing. Given the changes have been applied to a component used in multiple places, it would be helpful if you'd check the 
- `Recent activity` table
Screenshot 2024-11-21 at 12 19 46
- `Reputation` chart
Screenshot 2024-11-21 at 12 21 21
Screenshot 2024-11-21 at 12 21 29

- `Agreements` page
Screenshot 2024-11-21 at 12 20 38
Screenshot 2024-11-21 at 12 20 49

- `Balances` page
Screenshot 2024-11-21 at 12 20 10
Screenshot 2024-11-21 at 12 20 26

- `Members` page 
Screenshot 2024-11-21 at 12 24 48
Screenshot 2024-11-21 at 12 24 59

- `Permissions` page
Screenshot 2024-11-21 at 12 20 59
Screenshot 2024-11-21 at 12 21 10

- Open an existing action
Screenshot 2024-11-21 at 12 19 59

https://github.com/user-attachments/assets/c3496170-b4f9-4827-bef8-19b7eb08e66f

## Diffs

**Changes** 🏗

* Replace usage of `Portal` with `useDropdown` for `MeatBallMenu` component

Resolves #3579

### CODE REVIEW

**Review by iamsamgibbs — APPROVED** [review]
Great job @mmioana ! This works much better now.

https://github.com/user-attachments/assets/f1abe6f9-e591-46ad-8735-65aca88d1f25

There are a few differences compared to master, but I don't think any of them are a big deal. I'll list them here just so it's clear.

**Activity feed**
Placement is different
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 21 49  | Screenshot 2024-11-22 at 10 34 08 |

**Agreements**
Width is different
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 23 00  | Screenshot 2024-11-22 at 10 18 12 |

**Agreements mobile**
Width and left alignment is different (although this is probably wrong on master too?)
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 23 06 | Screenshot 2024-11-22 at 10 18 22 |

**Balance**
Placement is different
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 23 27 | Screenshot 2024-11-22 at 10 18 36 |

**Balance mobile**
Width and left alignment is different
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 23 58 | Screenshot 2024-11-22 at 10 18 44 |

**Members**
Width is different
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 24 19 | Screenshot 2024-11-22 at 10 19 07 |
 
**Members mobile**
Width and left alignment is different
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 24 30 | Screenshot 2024-11-22 at 10 19 20 |

**Permissions**
Width is different
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 24 56 | Screenshot 2024-11-22 at 10 19 37 |

**Permissions member**
Width is different (placement is slightly different)
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 25 01 | Screenshot 2024-11-22 at 10 19 43 |

**Permissions member mobile**
Width and left alignment is different
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 25 19 | Screenshot 2024-11-22 at 10 20 07 |

**Action meatball**
Placement is different
| This PR | Master |
| ------------- | ------------- |
| Screenshot 2024-11-22 at 10 25 37 | Screenshot 2024-11-22 at 10 20 28 |

File: src/components/v5/shared/MeatBallMenu/MeatBallMenu.tsx [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
 import clsx from 'clsx';
-import React, { type FC } from 'react';
+import React, { useRef, type FC } from 'react';
 
-import useRelativePortalElement from '~hooks/useRelativePortalElement.ts';
-import useToggle from '~hooks/useToggle/index.ts';
+import { useMobile } from '~hooks';
 import { formatText } from '~utils/intl.ts';
-import Portal from '~v5/shared/Portal/index.ts';
+import useDropdown from '~v5/frame/ColonyHome/partials/ReputationChart/hooks/useDropdown.ts';
```
Comment (bassgeta) [inline src/components/v5/shared/MeatBallMenu/MeatBallMenu.tsx:7]: should we extract this one to the `hooks` directory :D ?

File: src/components/v5/shared/MeatBallMenu/partials/MeatBallMenuItems.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          key,
+          label,
+          onClick,
+          icon: Icon,
+          renderItemWrapper: itemRenderItemWrapper,
+          className: itemClassName,
+        }) => (
+          <li key={key} className={clsx(itemClassName, 'flex-shrink-0')}>
+            <HoverWidthWrapper hoverClassName="w-full md:font-medium">
+              {(itemRenderItemWrapper || renderItemWrapper)(
```
Comment (bassgeta) [inline src/components/v5/shared/MeatBallMenu/partials/MeatBallMenuItems.tsx:26]: How I wish these were composed in instead :melting_face: But not in scope of this issue, would probably be way too much of a rework :D

**Review by bassgeta — APPROVED** [review]
GOod job solving this mess, the popups now work much much better, and one step towards better popup consolidation since we use like 3 approaches now :D 
1. Recent activity table :heavy_check_mark:  (do we maybe want to center it on mobile? seems to overflow the page actually...)
image
image
2. Reputation chart :heavy_check_mark: 
image
image
3. Agreements :heavy_check_mark: 
image
image
4. Balances :heavy_check_mark: 
image
image
5. Members :heavy_check_mark: 
image
image
7. Permissions page :heavy_check_mark: 
image
image

maybe it's more of an UX question, but should we actually just make the meatball menus be center aligned? seems like they overflow the page on normal phone screen sizes

but reapproving nonetheless, that can be tweaked separately :D

**Review by rdig — APPROVED** [review]
All the meatball menus that I've tested, both mobile and desktop work as expected. Very nicely done!


https://github.com/user-attachments/assets/b59ee5ae-17d4-45e2-b6bf-23ec1244b9a6


https://github.com/user-attachments/assets/6a6bfac4-6c74-4c3a-83a2-3d15a1bb540f

**Pull request merged**